libs/libc/machine/risc-v/common/arch_elf.c

```diff
@@ -224,7 +246,7 @@ bool up_checkarch(FAR const Elf_Ehdr *ehdr)
 
   if ((ehdr->e_entry & 1) != 0)
     {
-      berr("ERROR: Entry point is not properly aligned: %08" PRIx32 "\n",
+      berr("ERROR: Entry point is not properly aligned: %08lx\n",
```

Comment:
As soon as #2863 is merged, this change won't be needed.
We should stick to using the definitions from `inttypes.h`

Reply:
#2863 has been merged, @hotislandn

Reply:
PRIx32 works for RV32, but not for RV64:

machine/risc-v/common/arch_elf.c:241:12: warning: format '%x' expects argument of type 'unsigned int', but argument 3 has type 'Elf64_Addr' {aka 'const long unsigned int'} [-Wformat=]
       berr("ERROR: Entry point is not properly aligned: %08" PRIx32 "\n",
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
            ehdr->e_entry);
            ~~~~~~~~~~~~~

This parameter, "ehdr->e_entry" is also 'const long unsigned int' for RV32, which can be tested via following:

machine/risc-v/common/arch_elf.c:241:12: warning: format '%x' expects argument of type 'unsigned int', but argument 3 has type 'Elf32_Addr' {aka 'const long unsigned int'} [-Wformat=]
       berr("ERROR: Entry point is not properly aligned: %08" PRIxPTR "\n",
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
            ehdr->e_entry);
            ~~~~~~~~~~~~~

For "PRIx32", it is expanded to "lx" on RV32, but "x" on RV64; a similar issue happens for "PRIxPTR".

The reason for this change is that the input parameter has the same base type: 'const long unsigned int', which requires to use "lx" for printf, though it presents different data lengths on RV32 and RV64.